.github/workflows/run-kata-coco-tests.yaml

```diff
@@ -87,6 +88,7 @@ jobs:
           - nydus
         pull-type:
           - guest-pull
+          - host-share-image-block
```

Comment:
Looking at https://github.com/kata-containers/kata-containers/pull/7837 , it seems that this PR is in preparation to host share image-block with verity; so shouldn't the `pull-type` be something like `host-share-image-block-with-verity`?

Reply:
@wainersm Yes, #7837 is set up for host sharing, and the test covers only the verification of integrity support. It’s important to note, however, that this PR also support sharing without integrity checks. The presence of integrity verification in the block device is depended on the configuration settings of the nydus snapshotter. For example, configuring `export_mode` in the nydus snapshotter to `image_block_with_verity` enables integrity verification. Conversely, setting `export_mode` to `image_block` indicates that device integrity checks will not be performed.

Reply:
So I think `host-share-image-block` includes both `host-share-image-block-with-verity` and `host-share-image-block-without-verity`.

Reply:
Based on my understanding, the back-end is the same as guest-pull.

Reply:
>In general I'm concerned about these variations of configurations and if a single variable pull-type is enough to represent it in a scalable way.

Understanding your concern, could you tell me more about it?

Reply:
Hi @ChengyuZhu6 !

Please, take my words with caution because I'm still learning so I might say something that doesn't make sense.

> @wainersm Yes, #7837 is set up for host sharing, and the test covers only the verification of integrity support. It’s important to note, however, that this PR also support sharing without integrity checks. The presence of integrity verification in the block device is depended on the configuration settings of the nydus snapshotter. For example, configuring `export_mode` in the nydus snapshotter to `image_block_with_verity` enables integrity verification. Conversely, setting `export_mode` to `image_block` indicates that device integrity checks will not be performed.

ok. So setting `pull_type="host-share-image-block"` implies nydus-snapshotter's `fs_driver = "blockdev"` that uses https://github.com/containerd/nydus-snapshotter/blob/main/misc/snapshotter/config-blockdev.toml that has `export_mode = "image_block_with_verity"`,i.e., by default it will host-share blockdev with dm-verity. Now, to set the case of host-share blockdev **without** dm-verity: what's the plan?

Reply:
>First and foremost: is the case 3 supported by CoCo?
Yes, the case three is supported by both CoCo and Kata.

Reply:
>If case 3 and 4 are supported and pull-type is the same as 2 then how to deploy nydus-snapshotter for these cases? I also realized that [snapshotter.sh](https://github.com/containerd/nydus-snapshotter/blob/main/misc/snapshotter/snapshotter.sh) doesn't handle that cases

It depends on the value of `export_mode`. When set to `image_block`, case 3 is operational; conversely, setting `export_mode` to `layer_block_with_verity` enables case 4.

Reply:
Unfortunately, currently the snapshotter is limited to handling a single `export_mode` at any given time. Consequently, setting `export_mode` to `image_block_with_verity` precludes support for case 3 and case 4.  😢

Reply:
>Is there any use case for cases 5 and 6 in CoCo?

Yes. In CoCo, we now support on-demand pulling nydus images (including encrypted images) into the guest with `fusedev` driver, a feature provided by image-rs.
feature code: https://github.com/confidential-containers/guest-components/pull/117
test: https://github.com/confidential-containers/guest-components/pull/469

Reply:
@wainersm It’s important to note that the processes of `image_block_with_verity` and `image_block` are fundamentally similar https://github.com/kata-containers/kata-containers/pull/7837/commits/572ea7f8f6da4aa8fb67c5be208e587fad1d8b8c. The key difference is that `image_block_with_verity` calls image-rs to mount the dm-device, whereas `image_block` doesn't require this step. Based on this, it’s reasonable to assume that if case 2 is working, case 3 should also work as expected.

Reply:
Talking in private with @ChengyuZhu6 I finally understood how to set a completely custom configuration file so that [snapshotter.sh](https://github.com/containerd/nydus-snapshotter/blob/main/misc/snapshotter/snapshotter.sh) doesn´t need to change. Here is the doc that I missed: https://github.com/containerd/nydus-snapshotter/blob/main/docs/setup_snapshotter_by_daemonset.md#same-filesystem-with-different-snapshotter-configuration-and-different-nydusd-configuration